### PR TITLE
Combine sends to one core

### PR DESCRIPTION
**Pull request by ifilippov**
Idea is that 40GB can be handled only by 2 send queues,
however one CPU core can handle both queues, there is
no need to use two CPU cores. We can't use two threads
at one core, so this commit added support of multiple
send queues to one nff_go_send routine.

Semantic changes:
    1) Multiple SetSenders to one port will be merged to one send.
    2) If inIndexNumber = 1 (no scheduler or no RSS) performance
       can be worse because one send can't handle both send queues
       without RSS balancing.
    3) inIndexNumber > 1 must be even because send simply change
       send queues 0->1, 1->0. If inIndexNumber is uneven it will
       lead to reordering.
    4) TX_QUEUE_NUMBER defines number of send queues per port.
       2 queues in enough for 40GB, however it should be checked
       at other network cards.

---

Performance was checked and changed as expected.

### CODE REVIEW

File: low/low.h [hunk 1]
```diff
@@ -393,12 +396,13 @@ void nff_go_KNI(uint16_t port, volatile int *flag, int coreId,
 	*flag = wasStopped;
 }
 
-void nff_go_send(uint16_t port, int16_t queue, struct rte_ring **in_rings, int32_t inIndexNumber, volatile int *flag, int coreId) {
+void nff_go_send(uint16_t port, struct rte_ring **in_rings, int32_t inIndexNumber, bool anyway, volatile int *flag, int coreId) {
```
[the remark this answers is not in the document]
Comment (ifilippov) [reply low/low.h]: It is the same all across framework

File: flow/flow.go [hunk 2]
```diff
@@ -757,8 +760,23 @@ func SetSender(IN *Flow, portId uint16) error {
 		return common.WrapWithNFError(nil, "Requested send port exceeds number of ports which can be used by DPDK (bind to DPDK).", common.ReqTooManyPorts)
 	}
 	createdPorts[portId].wasRequested = true
-	addSender(portId, createdPorts[portId].txQueuesNumber, finishFlow(IN), IN.inIndexNumber)
-	createdPorts[portId].txQueuesNumber++
+	if createdPorts[portId].sendRings == nil {
+		// We create sendRings for this send port. All later
```
[the remark this answers is not in the document]
Comment (ifilippov) [reply flow/flow.go]: How?

File: flow/flow.go [hunk 3]
```diff
@@ -757,8 +760,23 @@ func SetSender(IN *Flow, portId uint16) error {
...
+		// sends will be merged with this ring.
+		// We need Sendrings with maximum possible number of
+		// connection groups so we check this max velue from all ports.
+		var max int32 = 0
```
[the remark this answers is not in the document]
Comment (aregm) [reply flow/flow.go]: // To allow consequent sends to one port, we need to create a send ring for the first, and then all the consequent sends should be merged with already created send ring.
//  We need send rings with max possible number of connection groups, to be able to merge all the consequent sends, so we set max to the max value from all the ports.

File: flow/flow.go [hunk 4]
```diff
@@ -757,8 +760,23 @@ func SetSender(IN *Flow, portId uint16) error {
...
+		for i := range createdPorts {
+			if createdPorts[i].InIndex > max {
+				max = createdPorts[i].InIndex
+			}
+		}
+		createdPorts[portId].sendRings = low.CreateRings(burstSize*sizeMultiplier, max)
+		addSender(portId, createdPorts[portId].sendRings, IN.inIndexNumber)
+	}
+	// All sends to one port will be handled by one core. One core is enough for 40GB/s
```
[the remark this answers is not in the document]
Comment (aregm) [reply flow/flow.go]: For a typical 40 GB card, like Intel 710 series, one core should be able to handle all the TX without problems.

**Pull request merged**